apps/server/src/modules/learnroom/service/course-sync.service.ts

```diff
@@ -42,36 +51,38 @@ export class CourseSyncService {
 	}
 
 	private async synchronize(courses: Course[], group: Group, oldGroup?: Group): Promise<void> {
-		if (courses.length) {
-			const [studentRole, teacherRole] = await Promise.all([
-				this.roleService.findByName(RoleName.STUDENT),
-				this.roleService.findByName(RoleName.TEACHER),
-			]);
-			const students = group.users.filter((groupUser: GroupUser) => groupUser.roleId === studentRole.id);
-			const teachers = group.users.filter((groupUser: GroupUser) => groupUser.roleId === teacherRole.id);
-
-			const coursesToSync = courses.map((course) => {
-				course.syncedWithGroup = group.id;
-				if (oldGroup && oldGroup.name === course.name) {
-					course.name = group.name;
-				}
-				course.startDate = group.validPeriod?.from;
-				course.untilDate = group.validPeriod?.until;
-
-				if (teachers.length >= 1) {
-					course.students = students.map((user: GroupUser): EntityId => user.userId);
-					course.teachers = teachers.map((user: GroupUser): EntityId => user.userId);
-				} else {
-					course.students = [];
-				}
-
-				course.classes = [];
-				course.groups = [];
-
-				return course;
-			});
-
-			await this.courseRepo.saveAll(coursesToSync);
+		const [studentRole, teacherRole] = await Promise.all([
+			this.roleService.findByName(RoleName.STUDENT),
+			this.roleService.findByName(RoleName.TEACHER),
+		]);
+
+		const studentIds = group.users
+			.filter((user: GroupUser) => user.roleId === studentRole.id)
+			.map((student) => student.userId);
+		const teacherIds = group.users
+			.filter((user: GroupUser) => user.roleId === teacherRole.id)
+			.map((teacher) => teacher.userId);
+
+		for (const course of courses) {
+			course.syncedWithGroup = group.id;
+			course.startDate = group.validPeriod?.from;
+			course.untilDate = group.validPeriod?.until;
+			course.classes = [];
+			course.groups = [];
+			course.students = studentIds;
```

Comment:
please check if the last teacher was removed from the group in case of a full sync. If that is the case, all students should be removed and the previous teacher should stay in the course.  This was a requirement which cannot be removed, though it is not relevant for the partial sync.